### ISSUE TITLE
Why separate parameters?

### ISSUE THREAD

**Issue by lgqfhwy** [louis-she/center-loss.pytorch#2]
Hello, I am a beginner. So maybe my question is simple. Please forgive me.
In the main.py. The code:
```
    model = model_class(num_classes).to(device)

    trainables_wo_bn = [param for name, param in model.named_parameters() if
                        param.requires_grad and 'bn' not in name]
    trainables_only_bn = [param for name, param in model.named_parameters() if
                          param.requires_grad and 'bn' in name]

    optimizer = torch.optim.SGD([
        {'params': trainables_wo_bn, 'weight_decay': 0.0001},
        {'params': trainables_only_bn}
    ], lr=args.lr, momentum=0.9)
```
I want to know why you separate the parameters to two parts: trainables_wo_bn and trainables_only_bn,
In the optimizer, I always write:
`optimizer = torch.optim.SGD(model.parameters(), lr = args.lr, momentum=0.9)`
Did I do something wrong? Please help me, thank you so much!

**Comment by louis-she** [comment]
Weight_decay is actually L2 norm to punish the complexity of the model. I think BN shouldn't be counted as complexity of a model(it does the other thing), so just ignore the L2 punishment of BN layers.
